### DISCUSSION TITLE
DAB releated queries.

### DISCUSSION THREAD

**Discussion by Madhan-Rommala** [Azure/data-api-builder#1524] [Q&A]
Hi Guys, I had couple of questions related to Data API Builder.

1. The Data API Builder container image (version 0.5.34) was hosted within an Azure Container Instance (ACI) that I set up 40 days ago. On top of that, I created a couple endpoints on Azure SQL Objects. Now I'm wondering if there is any way to upgrade this container image with the most recent version 0.7.6. So I would like to understand, is there any workaround to upgrade DAB container image automatically. If not how to do manually?

2. In single Azure container instance, can we configure multiple databases connections at a time like Azure SQL SB  and Azure Cosmos DB.? else do need to create a seperate Azure container instance to host Azure Cosmos db?
	
3. I would like to understand the performance of data api builder generated Data API's. Is DAB supports large data entities ? If yes, is there any study document for performance test?

4. Currently I'm using JWT token in order to access data api endpoints. But would like to see any other alternate option like user name  & password in order to access those api's.
	
5. In postman client, Is there a way or  sample code to retrieve all the data from api's in postman client or python script.  currently it is fetching first 100 rows and for remaining records I need to use nextlink attribute.

**Comment by abhishekkumams** [comment]
**5.** The default size is set to 100, but if you want more rows you can use the filter `first` to get as many rows as you want.

For your case lets say total count of rows is less than 500, then you can use the below request to get all the rows

REST:
```
https://localhost:5001/api/TableX?$first=500
```

GraphQL:
```
{
  TableX(first: 500) {
    items{
      id
      name
    }
  }
}
```

**Comment by Aniruddh25** [comment]
Hi @Madhan-Rommala, great questions :)

1. Since you are hosting DAB in your own Azure Container Instances, perhaps the documentation here might help you to update them: https://learn.microsoft.com/en-us/azure/container-instances/container-instances-update
If you are using the [Database Connections Feature of Azure Static Web] Apps(https://learn.microsoft.com/en-us/azure/static-web-apps/database-overview), the update will happen automatically for you in due course of time since it uses DAB underneath the hood.

2. You could configure multiple database connections in a single Azure Container Instance, but I think you would need following steps: 

- Modify the docker script [here](https://github.com/Azure/data-api-builder/blob/main/Dockerfile) and build your own image so that multiple DAB instances will be running within the docker container.
- If you are using this(https://learn.microsoft.com/en-us/azure/data-api-builder/run-using-container) to run the container, you would need to provide different configuration files for each DAB instance, and ensure different ports are mapped to different processes.

Haven't tried it, but welcome your contributions and experience if you'd like to try it out.

3. Yes, DAB supports large entities, we can share a study document of the performance testing that we have done.

4. Your Client frontend Application could use any 3rd Party Auth Provider that accepts Username and password to authenticate the end user, and eventually send the JWT token to DAB for further validating the JWT before authorization kicks in.

Hope this helps, Feel free to post any follow up questions you might have.

Thanks
Aniruddh.

**Comment by Madhan-Rommala** [comment]
Thank you for your prompt responses and clarifications, @Aniruddh25 @abhishekkumams.  

Regarding #Q3, Can you share us the study document link of the performance testing.

I had one more question: 

6. since Data API Builder generates APIs in HTTP URLs by default. Is it possible to generate HTTPS data API urls using DAB?

Sorry for the stupid questions :) ,  but I'm new to Azure and want to learn about its new features and functionality.

**Reply by abhishekkumams** [reply]
how are you trying to run the Data API builder? are you trying it locally or docker?

**Reply by Madhan-Rommala** [reply]
I was configured Data API builder docker image in Azure container instance.

**Reply by abhishekkumams** [reply]
We have updated the Docs showing how you can use HTTPS endpoints when using container: https://github.com/MicrosoftDocs/data-api-builder-docs-pr/blob/main/data-api-builder/run-using-container.md
(under the NOTES section)
